src/goto-symex/renaming_level.h

```diff
@@ -84,6 +86,77 @@ struct symex_level2t : public symex_renaming_levelt
 {
   symex_level2t() = default;
   ~symex_level2t() override = default;
+
+  std::shared_ptr<current_namest> global_names;
+
+  /// Ages the generation of an ssa expression by one.
+  void increase_generation(const irep_idt l1_identifier, const ssa_exprt &lhs)
+  {
+    current_names.emplace(l1_identifier, std::make_pair(lhs, 0));
+    if(global_names != nullptr)
+      global_names->emplace(l1_identifier, std::make_pair(lhs, 0));
+
+    increase_generation(l1_identifier);
+  }
+
+  /// Allocates a fresh L2 name for the given L1 identifier, and makes it the
+  /// latest generation on this path. Does nothing if there isn't an expression
+  /// keyed by the l1 identifier.
+  void increase_generation(const irep_idt identifier)
+  {
+    // If we can't find the name in the local scope, don't increase the global
+    // even if it exists there.
+    auto current_names_iter = current_names.find(identifier);
+    if(current_names_iter == current_names.end())
+      return;
+
+    // If we have a global store, increment its generation count, then assign
+    // that new value to our local scope.
+    if(global_names != nullptr)
+    {
+      auto global_names_iter = global_names->find(identifier);
+      if(global_names_iter != global_names->end())
+      {
+        global_names_iter->second.second++;
+        current_names_iter->second.second = global_names_iter->second.second;
+      }
+    }
+    else
+    {
+      // Otherwise simply increase our local scope.
+      current_names_iter->second.second++;
+    }
+  }
+
+  /// Prints the differences between the global and local naming maps (if they
+  /// exist)
+  void print_differences(const std::string &addition)
```

Comment:
This should be marked as a debug function

Reply:
Probably surround with `#if 0` along with an `iostream` include (and use `std::cerr`, since it's debug output?)